BUILDING.md

```diff
@@ -1,47 +1,65 @@
 # Building with CMake
 
+This project encourages uses of up-to-date build tools and contemporary development environments.
+To build from sources, you need:
+
+- CMake 3.26 or higher
+- A C++ compiler that supports C++23 (or equivalent) which is just a bug fix of C++20
+- An implementation of the C++ Core Guidelines Support Library (GSL)
+
 ## Dependencies
 
 For a list of dependencies, please refer to [vcpkg.json](vcpkg.json).
 
+This project depends on the [Microsoft implementation of GSL](https://github.com/microsoft/GSL), which is available on all supported platforms.  We recommend that you use your favorite package manager to install it prior to building and installing the IFC SDK from source.  If you just want to build the project and try it without installing it, then there is no need to install the GSL, but in that case the build step is a bit more involved.
+
+On Windows platforms, we recommend using [vckpg](https://vcpkg.io/en/getting-started.html) as the C++ package manager.  Issue
+```sh
+$ ./vckpg install --triplet x64-windows ms-gsl
+```
+in the VCPKG directory, to install the GSL.  The above command assumes you are building for the `x64` target.  Replace that with the appropriate triplet if you are taregting a different platform like `arm64`.
```

Comment:
> This suggests that vcpkg is somehow not cross-platform,

Hmm. Walk me slowly through the logical steps so I can arrive at the same conclusion.

Reply:
Nowadays it's fortunately rare, but you can still occasionally find people who think that think vcpkg is a Windows only product merely by coming from Microsoft and I don't think the wording here helps that. I have personally never used classic mode, so I don't know the quality of user experience of that on *nix systems. Manifest mode fortunately makes it very clear that it's for cross-platform use by the way it integrates with CMake.

I also think it's there's too much info about vcpkg here. Manifest mode is very simple and it's clearly documented with the CI workflows serving as an example.

Reply:
> Nowadays it's fortunately rare, but you can still occasionally find people who think that think vcpkg is a Windows only product merely by coming from Microsoft and I don't think the wording here helps that. 

How would you have phrased it?   Keep in mind that this project does not require VCPKG and it is not a goal.  Nor it is its mission to do VCPKG advocacy, or  make it vcpkg-centric.  It will take advantage of it where available.  

I think using classic mode is fine, even if someone of us don't personally use them.

> Manifest mode is very simple and it's clearly documented with the CI workflows serving as an example.

That might be true but the goal here isn't to be as minimal as possible, but to get people quickly started.  Repetition in that context isn't necessarily bad.  The experts will ignore those, and that is OK.

Reply:
If minimalism is preferable, I'd only document manifest mode, since that only requires passing one (two if you count target triplet) additional flag to CMake.

I'd also not mention vcpkg in an OS specific context. The users will have to figure out how to fit things into their system either way or they can rely on the software distribution on their Linux distro of choice once 0.43.1 is tagged. The prior is highly context sensitive, so not worth documenting here at all.

Reply:
> If minimalism is preferable, I'd only document manifest mode, since that only requires passing one (two if you count target triplet) additional flag to CMake.

Like I said, the goal of this document isn't to be as minimal as possible - but rather how to get people started as quickly as possible.   While we can require them to sift through several links or documentations, we might as well make it easier for them to get started with this project.  I updated the doc after I got more requests for help to get started (and we have a public record of some newcomers getting confused with terse instructions).

> The users will have to figure out how to fit things into their system either way

Yes, they will have.  There is value in facilitating that.

Is the update to this doc causing any actual confusion?